projects/fsastorefrontlib/src/shared/util/helpers/pipe/currency-detector.pipe.ts

```diff
@@ -0,0 +1,46 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { CurrencyService, WindowRef } from '@spartacus/core';
+import { combineLatest, iif, Observable, of } from 'rxjs';
+import { mergeMap } from 'rxjs/operators';
+import { CURRENCY_CODE } from '../../../../core/general-config/defalut-general-config';
+
+@Pipe({ name: 'currencyDetector' })
+export class CurrencyDetectorPipe implements PipeTransform {
+  currentCurrency$: Observable<string> = this.currencyService.getActive();
+
+  constructor(
+    protected currencyService: CurrencyService,
+    protected winRef: WindowRef
+  ) {}
+
+  transform(entry: any): Observable<string> {
+    const entryCurrencyStream = [of(entry), this.currentCurrency$];
+    const entryIsCurrency = entry.type === CURRENCY_CODE;
+
+    return combineLatest(entryCurrencyStream).pipe(
+      mergeMap(([entry, currency]) =>
+        iif(
```

Comment:
`iif`?

Reply:
Quick reference to the "IIF" docs can tell you how it is used: https://rxjs.dev/api/index/function/iif
As far as I can see, this code is legit. :)

Reply:
> Quick reference to the "IIF" docs can tell you how it is used: https://rxjs.dev/api/index/function/iif As far as I can see, this code is legit. :)

Thanks, good to know. I would just then maybe omit putting this condition in constant, should be much easier to read, and it's not being used anywhere else in the code anyway.